config/src/main/java/com/typesafe/config/impl/ConfigNodeInclude.java

```diff
@@ -0,0 +1,31 @@
+package com.typesafe.config.impl;
+
+import java.util.Collection;
+
+final class ConfigNodeInclude extends ConfigNodeComplexValue {
+    ConfigNodeInclude(Collection<AbstractConfigNode> children) {
+        super(children);
+    }
+
+    protected String kind() {
+        for (AbstractConfigNode n : children) {
+            if (n instanceof ConfigNodeSingleToken) {
+                Token t = ((ConfigNodeSingleToken) n).token();
+                if (Tokens.isUnquotedText(t) && !t.tokenText().equals("include") &&
+                        t.tokenText().matches(".*\\w.*")) {
```

Comment:
yeah - this is kind of a mini-parsing step here. I think it'd be better to grab the kind in the parser and then pass it to the ConfigNodeInclude constructor or something. Uses a bit more memory but avoids having so much knowledge of syntax here in this method after the parser has completed.